packages/beacon-node/src/chain/chain.ts

```diff
@@ -853,6 +853,19 @@ export class BeaconChain implements IBeaconChain {
     if (headState) {
       this.opPool.pruneAll(headState, finalizedState);
     }
+
+    // Populate finalized pubkey cache
+    if (finalizedState?.epochCtx.isAfterEIP6110()) {
```

Comment:
after #5968 we may not have finalized state anymore, since finalized validators cache is shared we can just use `headState`?

Reply:
just notice that we need to access number of validators in new finalized state anyway so let's handle this later. For now drop some TODOs here:
`TODO: finalizedState may not be available after https://github.com/ChainSafe/lodestar/issues/5968 . In that case implement getCheckpointStateValidatorCount() in regen and use headState instead`